sw/device/silicon_creator/rom_exts/rom_ext_common.ld

```diff
@@ -80,14 +80,29 @@ SECTIONS {
     *(.rodata.*)
   } > eflash
 
+  /**
+   * Critical static data that is accessible by both the mask ROM and the ROM
+   * extension.
+   *
+   * Each variable added to .critical_static must be in its own subsection
+   * named after the variable.
+   *
+   * This data is not initialized during CRT.
+   */
+  .critical_static ORIGIN(ram_main) (NOLOAD) : ALIGN(4) {
+    ASSERT(. - ADDR(.critical_static) == 0, "Error: .critical_static.sec_mmio_ctx section offset has changed.");
+    KEEP(*(.critical_static.sec_mmio_ctx))
+    ASSERT(. - ADDR(.critical_static) == 816, "Error: .critical_static.sec_mmio_ctx section size has changed");
+  } > ram_main
```

Comment:
Maybe we should consider factoring this out and `INCLUDE`ing it?

Reply:
Yeah I think you are right. I added the inner part to a separate file and `INCLUDE`d it. I've left the section declaration in the individual linker files for now.